### PR TITLE
DR-922 - Send missing exit code when exiting the app + Refactor protractor configuration

### PR DESCRIPTION
**Pull request by gastonmancini**
### Background

Travis CI was succeeding even if the protractor tests fail on SauceLabs.

### Change

- Send the exit code on the close event. 
- Refactor protractor configuration.

### Pending to be done

- Revisit the SauceLabs badge that is showing "unknown" state. This error might be related wite the bug mentioned on [this comments](https://wiki.saucelabs.com/display/DOCS/Using+Status+Badges+and+the+Browser+Matrix+Widget+to+Monitor+Test+Results).

### CODE REVIEW

File: src/gulpfile.js [hunk 1]
```diff
@@ -422,7 +422,7 @@ gulp.task('test:travis', function (done) {
   child_process
     .spawn(getNodeBinary(), argv, { stdio: 'inherit' })
     .on('error', function (e) { throw e; })
-    .on('close', function (e) { process.exit(); });
+    .on('close', function (e) { process.exit(e); });
```
Comment (andresmoschini) [inline src/gulpfile.js]: Great! could you appy the same fix to `test:component`?

**Comment by andresmoschini** [comment]
Nice work with the refactoring @gastonmancini !!

Does it pass the tests? in that case :shipit: !

File: src/gulpfile.js (hunk 1 above)
Comment (gastonmancini) [inline src/gulpfile.js]: Done!

**Comment by arzafran** [comment]
approved

**Comment by gastonmancini** [comment]
Thanks @arzafran!

**Pull request merged**